### PR TITLE
[jsoncpp] set DEBUG_LIBNAME_SUFFIX=d

### PR DESCRIPTION
**Pull request by shelltdf**

### CODE REVIEW

**Comment by vicroms** [comment]
Hi @shelltdf 

Please update the CONTROL version when modifying a portfile.

File: ports/jsoncpp/portfile.cmake [hunk 1]
```diff
@@ -24,6 +24,7 @@ vcpkg_configure_cmake(
             -DBUILD_SHARED_LIBS:BOOL=${JSONCPP_DYNAMIC}
             -DJSONCPP_WITH_PKGCONFIG_SUPPORT:BOOL=OFF
             -DJSONCPP_WITH_TESTS:BOOL=OFF
+            -DDEBUG_LIBNAME_SUFFIX=d
```
Comment (vicroms) [inline ports/jsoncpp/portfile.cmake]: Update version number in the CONTROL file when modifying a port.

**Review by vicroms — CHANGES_REQUESTED** [review]

File: ports/jsoncpp/portfile.cmake (hunk 1 above)
Comment (vicroms) [reply ports/jsoncpp/portfile.cmake]: Also, this change causes regressions on Windows.
Try something like this:

```
# No suffix on Windows and UWP.
if (not VCPKG_CMAKE_SYSTEM_NAME or VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
    set(DEBUG_LIBNAME_SUFFIX)
else()
    set(DEBUG_LIBNAME_SUFFIX d)
endif()

...

vcpkg_configure_cmake(
    ...
    -DDEBUG_LIBNAME_SUFFIX=${DEBUG_LIBNAME_SUFFIX}
)
```

**Comment by NancyLi1013** [comment]
Hi @shelltdf, here are the latest test results from the current CI system:

|triplet           |port      |master| test |notes
|------------------|----------|------|------|-----
|       x64-windows|          |      |      |
|                  |    pdal-c|  Pass|  Skip|
|                  |      pdal|  Pass|  Fail|**Regression**
|x64-windows-static|          |      |      |No Change
|           x64-osx|          |      |      |No Change
|     arm64-windows|          |      |      |No Change
|       x86-windows|          |      |      |
|                  |    pdal-c|  Pass|  Skip|
|                  |      pdal|  Pass|  Fail|**Regression**
|         x64-linux|          |      |      |No Change
|           x64-uwp|          |      |      |No Change
|           arm-uwp|          |      |      |No Change

**Comment by Neumann-A** [comment]
please see #6014. Is there some reason why this is required?

**Comment by grdowns** [comment]
Hey @shelltdf, I'm going to close this PR for now due to lack of activity and unclear motivation for the changes. Feel free to reopen the PR or restart the conversation about these changes whenever it is best for you. Thanks for the contribution!

**Pull request closed without merging**